osquery/tables/system/darwin/user_groups.mm

```diff
@@ -55,6 +56,74 @@ void genODEntries(ODRecordType type, std::set<std::string> &names) {
   }
 }
 
+void genPolicyColumns(int uid, boost::property_tree::ptree& tree) {
+  @autoreleasepool {
+    ODSession* s = [ODSession defaultSession];
+    NSError* err = nullptr;
+    ODNode* root = [ODNode nodeWithSession:s name:@"/Local/Default" error:&err];
+    if (err != nullptr) {
+      TLOG << "Error with OpenDirectory node: "
+           << std::string([[err localizedDescription] UTF8String]);
+      return;
+    }
+
+    ODQuery* q = [ODQuery queryWithNode:root
+                         forRecordTypes:kODRecordTypeUsers
+                              attribute:kODAttributeTypeUniqueID
+                              matchType:kODMatchEqualTo
+                            queryValues:[NSString stringWithFormat:@"%d", uid]
+                       returnAttributes:@"dsAttrTypeNative:accountPolicyData"
+                         maximumResults:0
+                                  error:&err];
+    if (err != nullptr) {
+      TLOG << "Error with OpenDirectory query: "
+           << std::string([[err localizedDescription] UTF8String]);
+      return;
+    }
+
+    // Obtain the results synchronously, not good for very large sets.
+    NSArray* od_results = [q resultsAllowingPartial:NO error:&err];
+    if (err != nullptr) {
+      TLOG << "Error with OpenDirectory results: "
+           << std::string([[err localizedDescription] UTF8String]);
+      return;
+    }
+    for (ODRecord* re in od_results) {
+      if (err != nullptr) {
```

Comment:
Is this error check redundant with the check on line 86? I can't see anything that resets the error at the end of the loop.